examples/huggingface-transformers/server.py

```diff
@@ -0,0 +1,189 @@
+
+Host a BERT ONNX model as a server, using the DeepSparse Engine and Flask
+
+positional arguments:
+  onnx_filepath         The full filepath of the ONNX model file or SparseZoo
+                        stub to the model
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -c NUM_CORES, --num-cores NUM_CORES
+                        The number of physical cores to run the analysis on,
+                        defaults to all physical cores available on the system
+  -a ADDRESS, --address ADDRESS
+                        The IP address of the hosted model
+  -p PORT, --port PORT  The port that the model is hosted on
+
+##########
+Example command for running using a model from sparsezoo:
+python server.py \
+    zoo:nlp/question_answering/bert-base/pytorch/huggingface/squad/pruned-moderate
+"""
+import argparse
+import json
+import time
+from typing import Any, Callable, List
+
+import flask
+from flask import Flask, jsonify, make_response
+from flask_cors import CORS
+
+from deepsparse.transformers import pipeline
+
+
+app = Flask(__name__)
+CORS(app)
+
+
+def is_str_list_or_str(val):
+    if isinstance(val, str):
+        return True
+    if isinstance(val, List):
+        return all(isinstance(elem, str) for elem in val)
+    return False
+
+
+def run_server(
+    predictor: Callable[[Any, Any], Any],
+    host: str = "0.0.0.0",
+    port: str = "5543",
+    info: str = None,
+):
+    """
+    Method to create routes and serve predictor on the specified port
+    """
+    assert callable(predictor), "predictor should be callable"
+
+    @app.route("/predict", methods=["POST"])
+    def predict():
+        """
+        Expects data in json format with question and context
+
+        '{"question":"what's my name?","context":"my name is snorlax"}'
+
+        :returns: A json with results from predictor and the inputs
+        """
+        data = json.loads(flask.request.get_data())
+
+        assert "question" in data and "context" in data
+        assert is_str_list_or_str(data["question"]) and is_str_list_or_str(
+            data["context"]
+        )
+
+        start = time.time()
+        result = predictor(question=data["question"], context=data["context"])
+        inference_time = (time.time() - start) * 1000
+
+        print(f"inference time: {inference_time:.4f} ms")
+        return jsonify({"result": result, "inputs": data})
+
+    @app.route("/info", methods=["GET"])
+    def information():
+        """
+        Route for model information
+        """
+        return jsonify(
+            {
+                "model_path": info,
+                "engine": predictor.engine,
+            }
+        )
+
+    @app.route("/<page_name>")
+    def unimplemented(page_name: str):
+        """
+        Route for unimplemented pages
+        """
+        return make_response(
+            f"The page named {page_name} has not been implemented.",
```

Comment:
NIT, would be nice if we're overriding this to provide more info than what would normally be available such as to list out the routes that are callable